src/simularium/VisGeometry/types.ts

```diff
@@ -0,0 +1,40 @@
+import { Object3D } from "three";
+import PDBModel from "../PDBModel";
+import { InstancedMesh } from "../rendering/InstancedMesh";
+
+export interface MeshLoadRequest {
+    mesh: Object3D;
+    cancelled: boolean;
+    instances: InstancedMesh;
+}
+
+export enum GeometryDisplayType {
+    PDB = "PDB",
+    OBJ = "OBJ",
+    SPHERE = "SPHERE",
+    CUBE = "CUBE",
+    GIZMO = "GIZMO",
+}
+
+export type PrimitiveDisplayType =
+    | GeometryDisplayType.SPHERE
+    | GeometryDisplayType.CUBE
+    | GeometryDisplayType.GIZMO;
+
+export interface PDBGeometry {
+    geometry: PDBModel;
+    displayType: GeometryDisplayType.PDB;
+}
+
+export interface MeshGeometry {
+    geometry: MeshLoadRequest;
+    displayType: GeometryDisplayType.OBJ | PrimitiveDisplayType;
+}
+
+export type AgentGeometry = PDBGeometry | MeshGeometry;
```

Comment:
just looking at this, would it be cleaner if you just had 
```
export interface AgentGeometry {
  geometry: PDBModel | MeshLoadRequest;
  displayType: GeometryDisplayType;
}
```
?

Reply:
and then that starts to look like the GeometryStoreLoadResponse too

Reply:
Right, but I want to be able to type a return type as either PDBGeometry or MeshGeometry so that I can access attributes only on PDBModel or MeshLoadRequest without having to do a bunch of additional if statements

Reply:
For instance: https://github.com/allen-cell-animated/simularium-viewer/pull/185/files/572bcfc73001255d9e1f2bda1c6dd3294d4b5a70#diff-be85a101dc76608957bdc5a6860c22c7dc95827dc01a9974a8cf2980d3ca264fR90. 

I also think it's a more accurate typing since you should never have `displayType === PDB` and `geometry === MeshLoadRequest`

Reply:
i agree those shouldn't happen but typechecking is not going to stop that from happening ,as it's all coming from runtime data

Reply:
Yeah, I just was more thinking of the typing as also a form of documentation, and trying to have it be clear what the expected input data should be, not that it would stop the problem.

Reply:
I might do a PR just to see what the code looks like with this change :)